include/xeus-zmq/xserver_zmq.hpp

```diff
@@ -58,39 +40,25 @@ namespace xeus
         void stop_impl() override;
         void update_config_impl(xconfiguration& config) const override;
 
-        void poll(long timeout);
-        void start_publisher_thread();
-        void start_heartbeat_thread();
-        void stop_channels();
-
-        zmq::socket_t m_shell;
-        zmq::socket_t m_controller;
-        zmq::socket_t m_stdin;
-        zmq::socket_t m_publisher_pub;
-        zmq::socket_t m_publisher_controller;
-        zmq::socket_t m_heartbeat_controller;
-
-        using authentication_ptr = std::unique_ptr<xauthentication>;
-        authentication_ptr p_auth;
+    private:
 
-        publisher_ptr p_publisher;
-        heartbeat_ptr p_heartbeat;
-
-        xthread m_iopub_thread;
-        xthread m_hb_thread;
-
-        using trivial_messenger_ptr = std::unique_ptr<xtrivial_messenger>;
-        trivial_messenger_ptr p_messenger;
-
-        nl::json::error_handler_t m_error_handler;
-        
-        bool m_request_stop;
+        xserver_zmq_impl* p_impl;
```

Comment:
Storing a `std::unique_ptr<xserver_zmq_impl>` instead of a raw pointer would allow to use the default destructor.